### PR TITLE
add: `zap`

### PR DESCRIPTION
**Pull request by LordUbuntu**
Add initial pacstall script for zap appimage package manager. Related to [this issue](https://github.com/srevinsaju/zap/issues/82).

The script is very basic and just relies on the curl-based shell script that [zap](https://github.com/srevinsaju/zap) itself uses to setup and install itself.

### CODE REVIEW

File: packages/zap/zap.pacscript [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name="zap"
+version="2.2.1"
+build_depends="curl grep jq wget"
```
Comment (Henryws) [inline packages/zap/zap.pacscript:3]: These would be `depends` correct?

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/zap/zap.pacscript (hunk 1 above)
Comment (LordUbuntu) [reply packages/zap/zap.pacscript:3]: Yep, dependencies as specified [here](https://github.com/srevinsaju/zap#automated-installation)

Comment (LordUbuntu) [reply packages/zap/zap.pacscript]: Yes. All of them are required (though wget is optional).

File: packages/zap/zap.pacscript [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name="zap"
+version="2.2.1"
+depends="curl grep jq wget"
+description="A delightful AppImage package manager"
```
Comment (Henryws) [inline packages/zap/zap.pacscript:4]:
```suggestion
url="https://github.com/srevinsaju/zap/releases/download/v2.2.1/zap-amd64"
description="A delightful AppImage package manager"
```

**Review by Henryws — CHANGES_REQUESTED** [review]
Find the hash of this file

File: packages/zap/zap.pacscript [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name="zap"
+version="2.2.1"
+depends="curl grep jq wget"
+url="https://github.com/srevinsaju/zap/releases/download/v2.2.1/zap-amd64"
```
Comment (Henryws) [inline packages/zap/zap.pacscript:4]: Find the hash, and include that as described in the wiki

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/zap/zap.pacscript (hunk 3 above)
Comment (LordUbuntu) [reply packages/zap/zap.pacscript:4]: sha256sum hash has been recorded and added.

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/zap/zap.pacscript [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        true
+}
+
+# doesn't need to be built
+build() {
+        true
+}
+
+install() {
+        curl https://raw.githubusercontent.com/srevinsaju/zap/main/install.sh | sudo bash -s
```
[the remark this answers is not in the document]
Comment (LordUbuntu) [reply packages/zap/zap.pacscript:19]: > This line can be replaced with something like: `sudo cp zap-amd65 "$pkgdir/usr/bin/zap`

Bet, substituting `curl https://raw.githubusercontent.com/srevinsaju/zap/main/install.sh | sudo bash -s` with `sudo cp zap-amd65 "$pkgdir/usr/bin/zap"` now...

**Review by Henryws — CHANGES_REQUESTED** [review]
Add `repology` array as described in the wiki

**Review by Henryws — CHANGES_REQUESTED** [review]
Use correct tabbing

**Comment by LordUbuntu** [comment]
> Use correct tabbing

I'm not sure what the correct tabbing is supposed to be, I couldn't find any specifications regarding that.

**Comment by Henryws** [comment]
Ok so I checked, it's not currently in the wiki, but we generally use 4 spaces as a tab.

File: packages/zap/zap.pacscript [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+prepare() {
+	true
+}
+
+build() {
+	true
+}
+
+install() {
+	sudo cp zap-amd65 "$pkgdir/usr/bin/zap"
```
Comment (Henryws) [inline packages/zap/zap.pacscript:20]:
```suggestion
	sudo cp zap-amd64 "$pkgdir/usr/bin/zap"
```

**Review by Henryws — CHANGES_REQUESTED** [review]

**Comment by LordUbuntu** [comment]
> Ok so I checked, it's not currently in the wiki, but we generally use 4 spaces as a tab.

Alright, I'll do that.

BTW, should we make an issue about tabbing rules in the wiki?

**Comment by Henryws** [comment]
I already added a section to it after you brought it up 😉

**Review by Henryws — APPROVED** [review]
Works!

**Comment by LordUbuntu** [comment]
Awesome! Thank you so much for your help and suggestions!

**Pull request merged**